modules/general/app-configuration/main.bicep

```diff
@@ -45,6 +63,26 @@ resource app_config_store 'Microsoft.AppConfiguration/configurationStores@2020-0
   tags: resourceTags
 }
 
+module connection_string_secret '../key-vault-secret/main.bicep' = if (saveConnectionStringsToKeyVault) {
+  name: 'appConfigConnStrSecretDeploy${name}'
+  scope: resourceGroup(keyVaultSubscriptionId, keyVaultResourceGroupName) 
+  params: {
+    keyVaultName: keyVaultName
+    secretName: keyVaultConnectionStringSecretName
+    contentValue: listKeys(name, '2020-06-01').value[0].connectionString
```

Comment:
Will `name` resolve properly if the Key Vault is in a different resource group to the AppConfig store?

Maybe we can reference the AppConfig store directly instead? (also avoids having to keep the API versions in-sync)

```
var acsKey = useExisting ? existing_app_config_store.listKeys().value[0] : app_config_store.listKeys().value[0]
module ... {
  params: {
    contentValue: acsKey.connectionString
  }
}
```